lib/common/typeorm.utils.ts

```diff
@@ -1,13 +1,33 @@
 import { Logger } from '@nestjs/common';
 import { Observable } from 'rxjs';
 import { delay, retryWhen, scan } from 'rxjs/operators';
-import { Connection, ConnectionOptions, EntityManager } from 'typeorm';
+import { Connection, ConnectionOptions, EntityManager, Repository, AbstractRepository } from 'typeorm';
 import * as uuid from 'uuid/v4';
 
+/**
+ * This function generates an injection token for an Entity or Repository
+ * @param {Function} This parameter can either be an Entity or Repository 
+ * @returns {string} The Entity | Repository injection token
+ */
 export function getRepositoryToken(entity: Function) {
```

Comment:
While we’re at it; lets change the type of entity
```suggestion
export function getRepositoryToken(entity: Repository | AbstractRepository | Entity) {
```

Reply:
One thing about this. Entity is a decorator and I'm not sure that type hint will work as expected. Also I think most people create classes like so:
```
@Entity()
export class User {}
```

I'm still new to node and Typescript but I believe this will prevent people from passing their entities.

Reply:
@JoeFitzgibbons Oh yes, totally right. Forgot that TypeOrm uses decorators for that. So seems like type `Function` is the way to go.